python/lsst/analysis/drp/makeQaTractTables.py

```diff
@@ -0,0 +1,353 @@
+
+
+class MakeForcedQaTractTablesTaskConnections(BaseMakeQaTractTablesTaskConnections,
+                                             defaultTemplates={"coaddName": "deep"},
+                                             dimensions=("tract", "skymap")):
+    qaTractTable = cT.Output(
+        doc="The collated catalog of forced measurements with additional columns for QA added",
+        storageClass="DataFrame",
+        name="qaTractTable_forced",
+        dimensions=("tract", "band", "skymap"),
+    )
+
+
+class MakeForcedQaTractTablesTaskConfig(BaseMakeQaTractTablesTaskConfig,
+                                        pipelineConnections=MakeForcedQaTractTablesTaskConnections):
+    # We want the following to come from the *_meas catalogs as they reflect
+    # what happened in SFP calibration.
+    columnsToCopyFromMeas = pexConfig.ListField(
+        dtype=str,
+        default=["calib_", ],
+        doc=("List of string \"prefixes\" to identify the columns to copy.  All columns with names "
+             "that start with one of these strings will be copied from the *_meas catalogs into the "
+             "*_forced_src catalogs UNLESS the full column name contains one of the strings listed "
+             "in the notInColumnStrList config.")
+    )
+    # We want the following to come from the *_ref catalogs as they reflect
+    # the forced measurement states.
+    columnsToCopyFromRef = pexConfig.ListField(
+        dtype=str,
+        default=["detect_", "merge_peak_sky", "merge_measurement_", ],
+        doc=("List of string \"prefixes\" to identify the columns to copy.  All columns with names "
+             "that start with one of these strings will be copied from the *_ref catalogs into the "
+             "*_forced_src catalogs UNLESS the full column name contains one of the strings listed "
+             "in the notInColumnStrList config.")
+    )
+
+
+class MakeForcedQaTractTablesTask(BaseMakeQaTractTablesTask):
+    """Read in {coaddName}Coadd_obj tables and aggregate QA columns by tract.
+
+    The values of interest here are those from the forced catalogs (i.e.
+    "forced_src") and, in this case, certain columns from the "meas" and "ref"
+    catalogs that don't get propagated to the "forced_src" catalogs are added
+    to the persisted table, which has dataset type "qaTractTable_forced".
+
+    See BaseMakeQaTractTablesTask for full description.
+    """
+    MEASUREMENT_TYPE = "forced"
+
+    ConfigClass = MakeForcedQaTractTablesTaskConfig
+    _DefaultName = "makeForcedQaTractTablesTask"
+
+    def collateTable(self, inputCat, activeFilterName, haveFilter):
+        """Implementation of BaseMakeQaTractTablesTask.collateTable for forced
+        measurements.
+        """
+        colsToLoadList = None
+        refColsToLoadList = None
+        measColsToLoadList = None
+
+        measCat = inputCat["meas"][activeFilterName]
+        forcedCat = inputCat["forced_src"][activeFilterName]
+        if colsToLoadList is None:
```

Comment:
If you've just set this to None how is it ever not None?

Reply:
Arghh...good point!  I guess I need to move the setting to `None` to `run` and pass them in (here and below in the unforced’s version) .  Thanks for catching this!